titan-compiler/driver.lua

```diff
@@ -95,6 +95,15 @@ function driver.shared()
     return shared
 end
 
+function driver.cflags()
+    local uname = io.popen("uname"):read("*a")
```

Comment:
This module is now doing the same `io.popen` in two places, and forking these subprocesses potentially multiple times, when multiple modules are compiled. We should cache the results, because in larger compilations this extra forking will add up. 

(Also, sorry to comment on an issue that existed prior to this patch, but this in an opportunity to fix it: both `io.popen` calls are not closing the fd and leaving up to the GC to deal with it... which my experience in LuaRocks shows it's not a good idea. Need to `io.popen` to a separate variable, read it then close it explicitly.)

Reply:
No problem, I will fix it.

Reply:
Cool @hishamhm! When I wrote the `io.popen` in the first place I had a feeling that I should have split the operations, but still went for the easier way. Many thanks for sharing your LuaRocks experience that confirms the operations should be split.